client.go

```diff
@@ -498,6 +529,24 @@ func bitsToImportRequest(indexName string, frameName string, slice uint64, bits
 	}
 }
 
+func statusToNodeSlicesForIndex(status *Status, indexName string) map[uint64]*URI {
```

Comment:
it's always a red flag for me when i see something nested more than one (or maybe two) levels. i understand why you've nested this series of for loops so deeply, but it still bothers me. at the very least i would change `if index.Name == indexName {...}` to `if index.Name != indexName { continue }` to remove one level of nesting.

also, are you ok ignoring errors from `NewURIFromAddress`?

finally, even though this is an unexported function I think it would be a good idea to add a comment to explain what it's for.